tools/cldr-apps/src/main/java/org/unicode/cldr/web/VettingViewerQueue.java

```diff
@@ -351,6 +281,87 @@ public void done() {
+                getUsersChoice(sm), "Winning " + SurveyMain.getNewVersion());
+            progress.update("Got VettingViewer");
+            statusCode = Status.PROCESSING;
+            start = System.currentTimeMillis();
+            last = start;
+            n = 0;
+            vv.setProgressCallback(new VettingViewer.ProgressCallback() {
+                public String setRemStr(long now) {
+                    double per = (double) (now - start) / (double) n;
+                    rem = (long) ((maxn - n) * per);
+                    String remStr = ElapsedTimer.elapsedTime(now, now + rem) + " " + "remaining";
+                    if (rem <= 1500) {
+                        remStr = "Finishing...";
+                    }
+                    setStatus(remStr);
+                    return remStr;
+                }
+
+                @Override
+                public void nudge() {
+                    if (!myThread.isAlive()) {
+                        throw new RuntimeException("Not Running- stop now.");
+                    }
+                    long now = System.currentTimeMillis();
+                    n++;
+                    // System.err.println("Nudged: " + n);
+                    if (n > (maxn - 5)) {
+                        maxn = n + 10;
+                        if (!isSummary && n > gMax) {
+                            gMax = n;
+                        }
+                    }
+
+                    if ((now - last) > 1200) {
+                        last = now;
+                        if (n > 500) {
+                            progress.update(n, setRemStr(now));
+                        } else {
+                            progress.update(n);
+                        }
+                    }
+                }
+
+                @Override
+                public void done() {
+                    progress.update("Done!");
+                }
+            });
+
+            EnumSet<VettingViewer.Choice> choiceSet = EnumSet.allOf(VettingViewer.Choice.class);
+            if (usersOrg.equals(Organization.surveytool)) {
+                choiceSet = EnumSet.of(
+                    VettingViewer.Choice.error,
+                    VettingViewer.Choice.warning,
+                    VettingViewer.Choice.hasDispute,
+                    VettingViewer.Choice.notApproved);
+            }
+
+            if (!isSummary(locale)) {
+                vv.generateHtmlErrorTables(aBuffer, choiceSet, locale.getBaseName(), usersOrg, usersLevel, false);
+            } else {
+                if (DEBUG)
+                    System.err.println("Starting summary gen..");
+                /*
+                 * TODO: remove call to getLocalesWithVotes here, unless it has necessary side-effects.
+                 * Its return value was formerly an unused argument to generateSummaryHtmlErrorTables
+                 */
+                getLocalesWithVotes(st_org);
+                vv.generateSummaryHtmlErrorTables(aBuffer, choiceSet, usersOrg);
```

Comment:
does getLocalesWithVotes have necessary side-effects?

Reply:
I can't see any. THis should be a separate ticket, though.

Reply:
Filed https://unicode-org.atlassian.net/browse/CLDR-14537